tensorflow/core/kernels/data/finalize_dataset_op.cc

```diff
@@ -38,11 +38,26 @@ void GetModelDatasetParams(const Options& options,
                            model::AutotuneAlgorithm* algorithm,
                            bool* cpu_budget, bool* ram_budget) {
   *algorithm = model::AutotuneAlgorithm::HILL_CLIMB;
-  if (options.optimization_options().autotune_buffers()) {
-    *algorithm = model::AutotuneAlgorithm::GRADIENT_DESCENT;
+
+  // check for deprecated autotune options in optimization_options
+  // the condition below checks for the presence of the deprecated option
+  // and decides whether to go with the deprecated options or not.
+  try{
+    if(options.optimization_options().optional_autotune_buffers_case() ==
+      OptimizationOptions::kAutotuneBuffers){
```

Comment:
@jsimsa I am trying to check if the deprecated attrs are present in the `optimization_options` or not. However, I am unable to catch this error:

```console
tensorflow/core/kernels/data/finalize_dataset_op.cc:47:28: error: 'kAutotuneBuffers' is not a member of 'tensorflow::data::OptimizationOptions'
       OptimizationOptions::kAutotuneBuffers){
```

What is the recommended approach to handle such exceptions/errors?